reactor-core-micrometer/src/noContextPropagationTest/java/reactor/core/observability/micrometer/MicrometerObservationListenerTest.java

```diff
@@ -0,0 +1,139 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.observability.micrometer;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+import io.micrometer.observation.tck.ObservationContextAssert;
+import io.micrometer.observation.tck.TestObservationRegistry;
+import io.micrometer.observation.tck.TestObservationRegistryAssert;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import reactor.util.context.Context;
+import reactor.util.context.ContextView;
+
+import static io.micrometer.observation.tck.TestObservationRegistryAssert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Simon Baslé
+ */
+public class MicrometerObservationListenerTest {
+
+
+	TestObservationRegistry registry;
+	AtomicLong              virtualClockTime;
+	Clock                                      virtualClock;
+	MicrometerObservationListenerConfiguration configuration;
+	ContextView                                subscriberContext;
+
+	@BeforeEach
+	void initRegistry() {
+		registry = TestObservationRegistry.create();
+		virtualClockTime = new AtomicLong();
+		virtualClock = new Clock() {
+			@Override
+			public long wallTime() {
+				return virtualClockTime.get();
+			}
+
+			@Override
+			public long monotonicTime() {
+				return virtualClockTime.get();
+			}
+		};
+		configuration = new MicrometerObservationListenerConfiguration(
+			"testName",
+			KeyValues.of("testTag1", "testTagValue1","testTag2", "testTagValue2"),
+			registry,
+			false);
+		subscriberContext = Context.of("contextKey", "contextValue");
+	}
+
+	@Test
+	void whenNoPropagationGetsParentFromContext() {
+		configuration = new MicrometerObservationListenerConfiguration(
+			MicrometerObservationListener.ANONYMOUS_OBSERVATION,
+			//note: "type" key is added by MicrometerObservationListenerConfiguration#fromFlux (which is tested separately)
+			KeyValues.of("testTag1", "testTagValue1","testTag2", "testTagValue2"),
+			registry,
+			false);
+
+		//detached
+		TestObservationRegistry otherRegistry = TestObservationRegistry.create();
+		Observation parent = Observation.start("testParent", otherRegistry);
+		parent.openScope();
+
+		//attached to test registry
+		Observation inRegistry = Observation.start("inregistry", registry);
+		inRegistry.openScope();
+		Context contextWithAParent = Context.of(subscriberContext).put(Micrometer.OBSERVATION_CONTEXT_KEY, parent);
+
+		//NB: registry.getCurrentObservation() always reflects the last Observation.openScope, even if Observation was started from a different registry
```

Comment:
ThreadLocals :shrug: I mean there's clsoe to 0 probability that you will have 2 registries running at the same time